modules/rgbd/src/large_kinfu.cpp

```diff
@@ -0,0 +1,361 @@
+    if (_depth.type() != DEPTH_TYPE)
+        _depth.convertTo(depth, DEPTH_TYPE);
+    else
+        depth = _depth;
+
+    std::vector<MatType> newPoints, newNormals;
+    makeFrameFromDepth(depth, newPoints, newNormals, params.intr, params.pyramidLevels, params.depthFactor,
+                       params.bilateral_sigma_depth, params.bilateral_sigma_spatial, params.bilateral_kernel_size,
+                       params.truncateThreshold);
+
+    std::cout << "Current frameID: " << frameCounter << "\n";
+    for (const auto& it : submapMgr->activeSubmaps)
+    {
+        int currTrackingId = it.first;
+        auto submapData = it.second;
+        Ptr<Submap<MatType>> currTrackingSubmap = submapMgr->getSubmap(currTrackingId);
+        Affine3f affine;
+        std::cout << "Current tracking ID: " << currTrackingId << std::endl;
+
+        if(frameCounter == 0) //! Only one current tracking map
+        {
+            currTrackingSubmap->integrate(depth, params.depthFactor, params.intr, frameCounter);
+            currTrackingSubmap->pyrPoints = newPoints;
+            currTrackingSubmap->pyrNormals = newNormals;
+            continue;
+        }
+
+        //1. Track
+        bool trackingSuccess =
+            icp->estimateTransform(affine, currTrackingSubmap->pyrPoints, currTrackingSubmap->pyrNormals, newPoints, newNormals);
+        if (trackingSuccess)
+            currTrackingSubmap->composeCameraPose(affine);
+        else
+        {
+            std::cout << "Tracking failed" << std::endl;
+            continue;
+        }
+
+        //2. Integrate
+        if(submapData.type == SubmapManager<MatType>::Type::NEW || submapData.type == SubmapManager<MatType>::Type::CURRENT)
+        {
+            float rnorm = (float)cv::norm(affine.rvec());
+            float tnorm = (float)cv::norm(affine.translation());
+            // We do not integrate volume if camera does not move
+            if ((rnorm + tnorm) / 2 >= params.tsdf_min_camera_movement)
+                currTrackingSubmap->integrate(depth, params.depthFactor, params.intr, frameCounter);
+        }
+
+        //3. Raycast
+        currTrackingSubmap->raycast(currTrackingSubmap->cameraPose, params.intr, params.frameSize, currTrackingSubmap->pyrPoints[0], currTrackingSubmap->pyrNormals[0]);
+
+        currTrackingSubmap->updatePyrPointsNormals(params.pyramidLevels);
+
+        std::cout << "Submap: " << currTrackingId << " Total allocated blocks: " << currTrackingSubmap->getTotalAllocatedBlocks() << "\n";
+        std::cout << "Submap: " << currTrackingId << " Visible blocks: " << currTrackingSubmap->getVisibleBlocks(frameCounter) << "\n";
+
+    }
+    //4. Update map
+    bool isMapUpdated = submapMgr->updateMap(frameCounter, newPoints, newNormals);
+
+    if(isMapUpdated)
+    {
+        // TODO: Convert constraints to posegraph
+        PoseGraph poseGraph = submapMgr->MapToPoseGraph();
+        std::cout << "Created posegraph\n";
+        Optimizer::optimizeCeres(poseGraph);
```

Comment:
Please change the name to `optimize()`, i.e. w/o mentioning a backend's name. This can be changed later.

Reply:
Done.